compile error on Fedora 28

uname -r
4.16.13-300.fc28.x86_64
cat /etc/redhat-release
Fedora release 28 (Twenty Eight)
when run
**"./autogen.sh && ./configure --prefix=/usr && make"**
get below error messages:

CC       obfs_local-encrypt.o
  CC       obfs_local-netutils.o
netutils.c: In function ‘setinterface’:
netutils.c:88:5: error: ‘strncpy’ specified bound 16 equals destination size [-W       error=stringop-truncation]
     strncpy(interface.ifr_name, interface_name, IFNAMSIZ);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:554: obfs_local-netutils.o] Error 1
make[2]: Leaving directory '/root/simple-obfs/src'
make[1]: *** [Makefile:406: all-recursive] Error 1
make[1]: Leaving directory '/root/simple-obfs'
make: *** [Makefile:338: all] Error 2

but shadowsocks-libev compile successful,  same vps Fedora 27 same compile procedure compile successful.

--- a/src/netutils.c
+++ b/src/netutils.c
@@ -85,7 +85,8 @@ setinterface(int socket_fd, const char *interface_name)
 {
     struct ifreq interface;
     memset(&interface, 0, sizeof(struct ifreq));
-    strncpy(interface.ifr_name, interface_name, IFNAMSIZ);
+    strncpy(interface.ifr_name, interface_name, IFNAMSIZ - 1);
+    interface.ifr_name[IFNAMSIZ - 1] = '\0';
     int res = setsockopt(socket_fd, SOL_SOCKET, SO_BINDTODEVICE, &interface,
                          sizeof(struct ifreq));
     return res;

